Updating patch to enable flag mcode-object-version=none

renjithravindrankannath opened the pull request:
Updating patch to enable flag mcode-object-version=none for device-libs when it is built as part of llvm-amdgpu

renjithravindrankannath committed:
Updating patch to add flag mcode-object-version=none when
device libs is buils as part of llvm-amdgpu

srekolam reviewed (changes requested):
i think the option  -Xclang -mcode-object-version=none is applicable for only +rocm-device-libs variant ..in the other case ,it is already doing that. So please have this patch applicable for only +rocm-device-libs .
Also, please add a comment so that this is understood as to why this patch is required. the patch name doesnot provide much info on the need.

renjithravindrankannath committed:
Limiting patch to +rocm-device-libs variant and adding
appropriate comment for the patch

renjithravindrankannath committed:
Updating llvmpatch as per the mailine code
Updating hsa-rocr patch as per the latest code
Updating the if elif condition for the hip test src path

renjithravindrankannath committed:
Updating flags for 5.5 relases and above

renjithravindrankannath committed:
Updating build flags and patches

renjithravindrankannath commented:
Hi @tldahlgren , Could you please review and approve if it looks good.

srekolam commented:
@tldahlgren can you please review this PR . Thanks.

tldahlgren reviewed (approved):
LGTM

srekolam reviewed (commented):
this PR mistakenly removed def cmake_args(self) in hsakmt-roct/package.py
